GVFS/GVFS.Common/NuGetUpgrade/ModernNugetUpgrader.cs

```diff
@@ -0,0 +1,204 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+
+namespace GVFS.Common.NuGetUpgrade
+{
+    public class ModernNugetUpgrader : NuGetUpgrader
+    {
+        private IQueryGVFSVersion gvfsVersionFetcher;
+
+        public ModernNugetUpgrader(
+           string currentVersion,
+           ITracer tracer,
+           PhysicalFileSystem fileSystem,
+           IQueryGVFSVersion gvfsVersionFetcher,
+           bool dryRun,
+           bool noVerify,
+           ModernNuGetUpgraderConfig config,
+           string downloadFolder,
+           ICredentialStore credentialStore)
+           : base(
+               currentVersion,
+               tracer,
+               fileSystem,
+               dryRun,
+               noVerify,
+               config,
+               downloadFolder,
+               credentialStore)
+        {
+            this.gvfsVersionFetcher = gvfsVersionFetcher;
+        }
+
+        public override bool SupportsAnonymousVersionQuery { get => true; }
+
+        private ModernNuGetUpgraderConfig Config { get => this.nuGetUpgraderConfig as ModernNuGetUpgraderConfig;  }
+
+        public static bool TryCreate(
+            ITracer tracer,
+            PhysicalFileSystem fileSystem,
+            LocalGVFSConfig gvfsConfig,
+            HttpClient httpClient,
+            ICredentialStore credentialStore,
+            bool dryRun,
+            bool noVerify,
+            out ModernNugetUpgrader upgrader,
+            out string error)
+        {
+            ModernNuGetUpgraderConfig upgraderConfig = new ModernNuGetUpgraderConfig(tracer, gvfsConfig);
+            upgrader = null;
+
+            if (!upgraderConfig.TryLoad(out error))
+            {
+                upgrader = null;
+                return false;
+            }
+
+            if (!upgraderConfig.IsConfigured(out error))
+            {
+                return false;
+            }
+
+            if (!upgraderConfig.IsReady(out error))
+            {
+                return false;
+            }
+
+            if (!TryParseOrgFromNugetFeedUrl(upgraderConfig.FeedUrl, out string orgName))

```

Comment:
Why is this still needed? I thought the config would now contain the org URL?

Reply:
The config contains the URL to the endpoint to query for available VFSForGit versions, but it does not (currently) know which organization it is querying for. This is a parameter to the query... it doesn't have to come from the NuGetFeed URL, but that is where it currently lives...

I can add more documentation on how the pieces fit tother here.